tests/datasource/fluent/integration/test_sql_datasources.py

```diff
@@ -43,12 +57,85 @@ def context() -> EphemeralDataContext:
     return ctx
 
 
+class TableFactory(Protocol):
+    def __call__(
+        self,
+        engine: engine.Engine,
+        table_names: set[str],
+        schema: str | None = None,
+    ) -> None:
+        ...
+
+
+@pytest.fixture(scope="function")
+def capture_engine_logs(caplog: pytest.LogCaptureFixture) -> pytest.LogCaptureFixture:
+    """Capture SQLAlchemy engine logs and display them if the test fails."""
+    caplog.set_level(logging.INFO, logger="sqlalchemy.engine")
+    return caplog
+
+
+@pytest.fixture(scope="function")
+def table_factory(
+    capture_engine_logs: pytest.LogCaptureFixture,
+) -> Generator[TableFactory, None, None]:
+    """
+    Given a SQLALchemy engine, table_name and schema,
+    create the table if it does not exist and drop it after the test.
+    """
+    all_created_tables: dict[
+        str, list[dict[Literal["table_name", "schema"], str | None]]
+    ] = {}
+    engines: dict[str, engine.Engine] = {}
+
```

Comment:
This could be simplified, originally this was going to be a module/class-scoped fixture that setup tables for multiple backends.